src/Servers/Kestrel/test/InMemory.FunctionalTests/RequestTests.cs

```diff
@@ -2262,4 +2273,13 @@ private class IntAsClass
             public int Value;
         }
     }
+
+    // Ensure that all common tests are still passing when the AcceptLineFeedAsLineTerminator quirk mode is enabled.
+    public class RequestTestsQuirkMode : RequestTests
```

Comment:
I do think it is excessive to run all or even most of the `HttpParserTests`, `BadHttpRequestTests` and `RequestsTests` in quirks mode. It's not really adding much coverage and it adds a lot of noise to the test results even if they're fast. And even if they are fast, it could be the start of a death by 1,000 paper cuts. Or should I say 2,000 paper cuts since this doubling what were already over 2000 tests? 😆 

As long as we share all the parsing logic between quirk mode and non-quirk mode aside from finding the line terminator, I think we need just a few new tests with EnableLineFeedTerminator:

1. Verify using just `\n` for the start line, all headers, and end of headers works
2. Verify using just `\r\n` for the start line, all headers, and end of headers works
3. Verify using a combination of `\r` and `\r\n` for the start line, all headers, and end of headers works
4. Verify that you can still get a bad request for something like an unexpected `\r` in the middle of the line.
5. Verify that `a:b\n` works since that's the shortest valid header.

Reply:
We'll remove all these extraneous test variations, right?

Reply:
Done